src/backend/marsha/core/views.py

```diff
@@ -397,6 +408,158 @@ class DocumentView(BaseLTIView):
+            raise PermissionDenied
+
+        try:
+            lti.verify()
+        except LTIException as error:
+            app_data = _manage_exception(error)
+        else:
+            app_data = self._get_app_data(lti)
+
+        return {
+            "app_data": json.dumps(app_data),
+            "static_base_url": f"{settings.STATIC_URL}js/",
+            "external_javascript_scripts": settings.EXTERNAL_JAVASCRIPT_SCRIPTS,
+        }
+
+    def _get_app_data(self, lti):
+        """Build app data for the frontend with information retrieved from the LTI launch request.
+
+        Returns
+        -------
+        dictionary
+            Configuration data to bootstrap the frontend:
+            A form is build in react to post data to the LTI Consumer thnough an iframe.
+
+            For instructors only
+            ++++++++++++++++++++
+
+            - lti_select_form_action_url: URL to post data to.
+            - lti_select_form_data: Data to post.
+            - new_document_url: LTI URL to a new document.
+            - new_video_url: LTI URL to a new video.
+            - documents: Documents list with theirs LTI URLs.
+            - videos: Videos list with theirs LTI URLs.
+
+        """
+        documents = DocumentSelectLTISerializer(
+            get_selectable_resources(Document, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        videos = VideoSelectLTISerializer(
+            get_selectable_resources(Video, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        new_uuid = str(uuid.uuid4())
+        app_data = _get_base_app_data()
+        app_data.update(
+            {
+                "frontend": "LTI",
+                "lti_select_form_action_url": reverse("select_lti_view"),
+                "lti_select_form_data": self.request.POST,
+                "new_document_url": self.request.build_absolute_uri(
+                    reverse("document_lti_view", args=[new_uuid])
+                ),
+                "new_video_url": self.request.build_absolute_uri(
+                    reverse("video_lti_view", args=[new_uuid])
+                ),
+                "documents": documents,
+                "videos": videos,
+            }
+        )
+        return app_data
+
+    # pylint: disable=unused-argument
+    def post(self, request, *args, **kwargs):
+        """Respond to POST request.
+
+        Populated with context retrieved by get_context_data in the LTI launch request.
+
+        Parameters
+        ----------
+        request : Request
+            passed by Django
+        args : list
+            positional extra arguments
+        kwargs : dictionary
+            keyword extra arguments
+
+        Returns
+        -------
+        HTML
+            generated from applying the data to the template
+
+        """
+        if "content_items" in self.request.POST:
```

Comment:
The docstring is not clear as it appears there are 2 very different scenarios here.
In one case you call `get_context_data` which `calls get_app_data`
In the other case you only call `get_base_app_data`
That's all I understand for now but it's confusing.
Maybe it requires more explanation or maybe some refactoring to treat these 2 scenarios in different views?

Reply:
Indeed, I was thinking about splitting those scnarios, mainly because when we sign the post request we want to post it back to the consumer.
I propose to use a django template for this purpose where we can build a form and submit it with simple vanilia js.
WDYT ?

Reply:
I don't know because I don't understand the second scenario to call this view :shrug:  
The first scenario is the consumer calling the select view with an LTI request. What is the second scenario and who calls it?

Reply:
The second scenario matches points 3 and 4 in the PR description :

> 3. When the user selects a content, it is JSON serialized to a content_items parameter and added to the form:
> 
> 4. The form is send to marsha.tld/lti/select/ wiith content_items.
> Marsha's backend filters those parameters, signs it with oauth, and sends back everything to the frontend application along with an lti_content_selected boolean in app context parameters and the LTI consumer url to respond to.

Reply:
Using a separate django view to handle this would allow to use it on another use case, and would simplify the react component I built.

Reply:
oh ok! Then yes it should be a completely different view as it is not an lti request.